frontend/src/app/containers/UpgradeWarningPage/index.js

```diff
@@ -0,0 +1,96 @@
+// @flow
+
+import React from 'react';
+import { Localized } from 'fluent-react/compat';
+
+import config from '../../config';
+
+import View from '../../components/View';
+import Copter from '../../components/Copter';
+import LayoutWrapper from '../../components/LayoutWrapper';
+import LocalizedHtml from '../../components/LocalizedHtml';
+
+type UpgradeWarningProps = {
+  sendToGA: Function,
+  host: string,
+  protocol: string,
+  isMinFirefox: Boolean,
+}
+
+export default class UpgradeWarning extends React.Component {
+  props: UpgradeWarningProps
+
+  componentWillMount() {
+    this.props.sendToGA('event', {
+      eventCategory: 'HomePage Interactions',
+      eventAction: 'Upgrade Warning',
+      eventLabel: 'upgrade notice shown'
+    });
+  }
+
+  renderTitle() {
+    const { isMinFirefox, protocol, host } = this.props;
+    let title = (<Localized id="warningGenericTitle">
+                   <span>Something is wrong!</span>
+                 </Localized>);
+    if (!isMinFirefox) {
+      title = (<Localized id="warningUpgradeFirefoxTitle">
+                 <span>Upgrade Firefox to continue!</span>
+               </Localized>);
+    } else if (protocol !== 'https:') {
+      title = (<Localized id="warningHttpsRequiredTitle">
+                 <span>HTTPS required!</span>
+               </Localized>);
+    } else if (config.devHosts.includes(host)) {
```

Comment:
Would be better to move tests involving `config` properties into the App component - e.g. [around where it's included in the page](https://github.com/mozilla/testpilot/pull/2918/files#diff-d1e3fb4bae03a0b6b298c56e7110d06fR167) or in [shouldShowUpgradeWarning](https://github.com/mozilla/testpilot/pull/2918/files#diff-d1e3fb4bae03a0b6b298c56e7110d06fR150). That way, this warning component can be just pure display driven by prop values. Should make writing stories easier, if only props need to be altered to step through display states.